airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/component/DataCoercionSuite.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.component
+
+import io.airbyte.cdk.load.data.AirbyteValue
+import io.airbyte.cdk.load.data.FieldType
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.NumberType
+import io.airbyte.cdk.load.dataflow.transform.ValueCoercer
+import io.airbyte.cdk.load.message.Meta
+import io.airbyte.cdk.load.table.ColumnNameMapping
+import io.airbyte.protocol.models.v0.AirbyteRecordMessageMetaChange.Reason
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import kotlinx.coroutines.test.runTest
+
+/**
+ * The tests in this class are designed to reference the parameters defined in
+ * `DataCoercionFixtures.kt`. For example, you might annotate [`handle integer values`] with
+ * `@MethodSource("io.airbyte.cdk.load.component.DataCoercionIntegerFixtures#int32")`. See each
+ * fixture class for explanations of what behavior they are exercising.
+ *
+ * Note that this class _only_ exercises [ValueCoercer.validate]. You should write separate unit
+ * tests for [ValueCoercer.map]. For now, the `map` function is primarily intended for transforming
+ * `UnionType` fields into other types (typically `StringType`), at which point your `validate`
+ * implementation should be able to handle any StringValue (regardless of whether it was originally
+ * a StringType or UnionType).
+ */
+@MicronautTest(environments = ["component"], resolveParameters = false)
+interface DataCoercionSuite {
+    val coercer: ValueCoercer
+    val airbyteMetaColumnMapping: Map<String, String>
+        get() = Meta.COLUMN_NAMES.associateWith { it }
+    val columnNameMapping: ColumnNameMapping
+        get() = ColumnNameMapping(mapOf("test" to "test"))
+
+    val opsClient: TableOperationsClient
+    val testClient: TestTableOperationsClient
+
+    val harness: TableOperationsTestHarness
+        get() = TableOperationsTestHarness(opsClient, testClient, airbyteMetaColumnMapping)
+
+    /** Fixtures are defined in [DataCoercionIntegerFixtures]. */
+    fun `handle integer values`(
+        inputValue: AirbyteValue,
+        expectedValue: Any?,
+        expectedChangeReason: Reason?
+    ) = runTest {
+        harness.testValueCoercion(
+            coercer,
+            columnNameMapping,
+            FieldType(IntegerType, nullable = true),
+            inputValue,
+            expectedValue,
+            expectedChangeReason,
+        )
+    }
+
+    /** Fixtures are defined in [DataCoercionNumberFixtures]. */
+    fun `handle number values`(
+        inputValue: AirbyteValue,
+        expectedValue: Any?,
+        expectedChangeReason: Reason?
+    ) = runTest {
+        harness.testValueCoercion(
+            coercer,
+            columnNameMapping,
+            FieldType(NumberType, nullable = true),
+            inputValue,
+            expectedValue,
+            expectedChangeReason,
+        )
+    }
```

Comment:
Do we have any non numeric cases? String lengths / blob sizes maybe?

Reply:
forthcoming :P I wanted to run the int/number stuff past you before implementing everything else, since these (+ timestamps) are the annoying data types.

(though in parallel, I'm starting up https://github.com/airbytehq/airbyte/pull/70322/files + https://github.com/airbytehq/airbyte/pull/70321/files for all the other data types. Will probably just throw an issue in the backlog for unions though, given the weirdness with `map`)